lib/workers/repository/update/pr/automerge.ts

```diff
@@ -69,6 +69,12 @@ export async function checkAutoMerge(
       prAutomergeBlockReason: 'PlatformNotReady',
     };
   }
+  // Usually the PR will already be approved, this is a last resort in case the
+  // approval was lost for some reason
+  if (config.autoApprove && platform.approvePr) {
+    logger.debug('Auto-approving PR if needed before automerge');
+    await platform.approvePr(pr.number);
+  }
```

Comment:
Will this cause each PR to be approved on each run? i.e. one extra API call per PR per run?

Reply:
> Will this cause each PR to be approved on each run?

Only if they are not yet approved.

> i.e. one extra API call per PR per run?

Per auto-merge attempt, there will be one extra API call to verify whether the PR is not yet approved, and one more API call to approve it in case it's not yet approved.

Just remember:

1. this is only applicable for azure, gitlab, and gerrit
2. and for people who have `autoApprove: true`
3. and all other conditions in `automerge.ts` already passed (except `branchStatus`, which depends on the approval to be green)

Reply:
But if you believe it's not a good idea, I can find another way.